components/feature/media/src/test/java/mozilla/components/feature/media/middleware/LastMediaAccessMiddlewareTest.kt

```diff
@@ -214,50 +284,516 @@ class LastMediaAccessMiddlewareTest {
+        assertFalse(MediaTabStatus("url1", "url2", 1).hasMediaEnded())
+    }
+
+    @Test
+    fun `GIVEN data about media tab exists WHEN maybeUpdate is called for that tab THEN update the current data`() {
+        // Using two separate tabs to ensure only the intended one is updated.
+        val mediaTab1 = TabSessionState(content = ContentState("https://mozilla.com"), id = "44")
+        val mediaTab2 = TabSessionState(content = ContentState("https://mozilla.com/firefox"), id = "45")
+        val mediaTabsStatus = mutableMapOf<String, MediaTabStatus>().apply {
+            this[mediaTab1.id] = MediaTabStatus(mediaTab1.content.url, mediaTab1.content.url)
+            this[mediaTab2.id] = MediaTabStatus(mediaTab2.content.url, mediaTab2.content.url)
+        }
+        val updatedUrl = "https://firefox.com"
+
+        mediaTabsStatus.maybeUpdate(mediaTab1.id) { it.copy(updatedTabUrl = updatedUrl) }
+
+        assertEquals(2, mediaTabsStatus.size)
+        assertEquals(updatedUrl, mediaTabsStatus[mediaTab1.id]!!.updatedTabUrl)
+        assertEquals(mediaTab2.content.url, mediaTabsStatus[mediaTab2.id]!!.updatedTabUrl)
+    }
+
+    @Test
+    fun `GIVEN data about media tab exists WHEN maybeUpdate is called for other tab THEN don't update the current data`() {
+        // Using two separate tabs to ensure only the intended one is updated.
+        val mediaTab1 = TabSessionState(content = ContentState("https://mozilla.com"), id = "44")
+        val mediaTab2 = TabSessionState(content = ContentState("https://mozilla.com/firefox"), id = "45")
+        val mediaTabsStatus = mutableMapOf<String, MediaTabStatus>().apply {
+            this[mediaTab1.id] = MediaTabStatus(mediaTab1.content.url, mediaTab1.content.url)
+            this[mediaTab2.id] = MediaTabStatus(mediaTab2.content.url, mediaTab2.content.url)
+        }
+        val updatedUrl = "https://firefox.com"
+
+        mediaTabsStatus.maybeUpdate("otherTabId") { it.copy(updatedTabUrl = updatedUrl) }
+
+        assertEquals(2, mediaTabsStatus.size)
+        assertEquals(mediaTab1.content.url, mediaTabsStatus[mediaTab1.id]!!.updatedTabUrl)
+        assertEquals(mediaTab2.content.url, mediaTabsStatus[mediaTab2.id]!!.updatedTabUrl)
+    }
+
+    @Test
+    fun `GIVEN normal tabs exist WHEN ifNormalTab is called THEN use it`() {
+        // Using two separate tabs to ensure only the intended one is updated.
+        val normalTab1: TabSessionState = mock()
+        val normalTab2 = TabSessionState(content = mock(), id = "normalTab")
+        val context: MiddlewareContext<BrowserState, BrowserAction> = mock()
+        val state = BrowserState(tabs = listOf(normalTab1, normalTab2))
+        doReturn(state).`when`(context).state
+        var effectToVerify = ""
+
+        context.ifNormalTab(normalTab2.id) { effectToVerify = it.id }
+
+        assertEquals(normalTab2.id, effectToVerify)
+    }
+
+    @Test
+    fun `GIVEN private tabs exist WHEN ifNormalTab is called THEN use it`() {
```

Comment:
See question above. Given the name, I expected `ifNormalTab` to only run for normal tabs. Why does it apply to a private tab here?

Reply:
"normal" here would mean normal and private tabs as opposed to custom tabs.
Maybe just calling them by name is better and in the event this kind of method will be needed in the new implementation it could be named `ifNormalOrPrivateTab`?